.github/workflows/contracts.yaml

```
@@ -196,20 +215,80 @@ jobs:
           yarn run hardhat --network ${{ github.event.inputs.environment }} \
             etherscan-verify --license GPL-3.0 --force-license
 
+  # This job is responsible for publishing packages from `dapp-development`
+  # branch, which are slightly modified to help with the process of testing some
+  # features on the Threshold Token dApp. The job starts only if workflow gets
+  # triggered by the `workflow_dispatch` event on the branch `dapp-development`.
+  contracts-dapp-development-deployment-testnet:
+    needs: [contracts-build-and-test]
+    if: |
+      github.event_name == 'workflow_dispatch'
+        && github.ref == 'refs/heads/dapp-development'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "14.x"
+          cache: "yarn"
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Resolve latest contracts
+        run: yarn upgrade @keep-network/keep-core@${{ github.event.inputs.environment }}
+
+      - name: Deploy contracts
+        env:
+          CHAIN_API_URL: ${{ secrets.GOERLI_ETH_HOSTNAME_HTTP }}
+          CONTRACT_OWNER_ACCOUNT_PRIVATE_KEY: ${{ secrets.DAPP_DEV_GOERLI_ETH_CONTRACT_OWNER_PRIVATE_KEY }}
+          KEEP_CONTRACT_OWNER_ACCOUNT_PRIVATE_KEY: ${{ secrets.GOERLI_KEEP_ETH_CONTRACT_OWNER_PRIVATE_KEY }}
+        run: yarn deploy --network ${{ github.event.inputs.environment }}
+
+      - name: Bump up package version
+        id: npm-version-bump
+        uses: keep-network/npm-version-bump@v2
+        with:
+          environment: dappdev${{ github.event.inputs.environment }}
```

Could we try using underscores?

```suggestion
          environment: dapp_dev_${{ github.event.inputs.environment }}
```

---

What do you think about using `*development*` instead of just `*dev*`? This would match the package tag more closely.
```suggestion
          environment: dapp_development_${{ github.event.inputs.environment }}
```

---

> Could we try using underscores?
I did try it some time ago, I was getting `invalid semver version: ${string}`

> What do you think about using *development* instead of just *dev*? This would match the package tag more closely.
I chose `dev` rather than `development` to be consistent with how we name the `development` tagged packages. We name them with `-dev.x` suffix, for exapmple `@keep-network/random-beacon@2.0.0-dev.64`.

---

Please see https://github.com/keep-network/npm-version-bump/pull/3. Once it's merged we should be able to use `dapp-dev`.

---

[944d392](https://github.com/threshold-network/solidity-contracts/pull/119/commits/944d392512f9b2c3c78ee3885231b41cd17686a3)